src/Microsoft.AspNetCore.Mvc.Cors/Internal/CorsHttpMethodActionConstraint.cs

```diff
@@ -10,6 +10,7 @@
 
 namespace Microsoft.AspNetCore.Mvc.Cors.Internal
 {
+    // Don't casually change the name of this. We reference the full type name in ActionConstraintCache.
     public class CorsHttpMethodActionConstraint : HttpMethodActionConstraint
```

Comment:
Do you want to write a test in Mvc.Test for this? `Assert.NotNull(Type.GetType(ActionConstraintCache.CorsHttpMethodConstraintName))`.

Reply:
Yeah, maybe I'll move these tests to .Mvc.Test. That's a good idea.